---
title: "Comparing_RNAi_RNAi2"
author: "Patricka Williams-Simon"
date: "11/5/2021"
output: html_document
---

```{r}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(dplyr)
library(stringr)
library(sva)
library(ggplot2)
library(cowplot)
library(colorspace)
library(ggrepel)
library(tidyverse)
library(multcomp)
library(readxl)
theme_set(theme_cowplot())


```
This script is compare raw TT reads from RNAi1 to RNAi2


#Load samples 
```{r}

ThermDat1 <- read.csv(file="../ProcessedData/RNAi1_tracked_data.csv")
str(ThermDat1)
ThermDat1[1:5,]
ThermDat1$genotype  <- as.character(ThermDat1$genotype)


ThermDat2 <- read.csv(file="../ProcessedData/RNAi2_tracked_data.csv")
str(ThermDat2)
ThermDat2[1:5,]
ThermDat2$genotype  <- as.character(ThermDat2$genotype)

#add a col set id 
ThermDat1$set_id<- paste(ThermDat1$genotype, ThermDat1$cross, sep="_")
ThermDat2$set_id<- paste(ThermDat2$genotype, ThermDat2$cross, sep="_")

#load background data
RNAi_backg <- read_excel("../ProcessedData/HeatPlate_autotrack/RNAi_Bckg.xlsx")
str(RNAi_backg)
RNAi_backg[1:5,]

#assign column names
colnames(RNAi_backg) <-c("genotype", "backg_id", "Gname")

#combine data sets
ThermDat1 <- left_join(ThermDat1, RNAi_backg, by="genotype")
ThermDat2 <- left_join(ThermDat2, RNAi_backg, by="genotype")



```

#make plot for each gene of interest 



```{r}

#pull data for each gene from RNAi1 and 2.
#ATPalpha

foc.gene <- 51411

FGene1 <- ThermDat1 %>% filter(genotype ==foc.gene)
FGene2 <- ThermDat2 %>% filter(genotype ==foc.gene)

Bk1 <- ThermDat1 %>% filter(genotype == ifelse(FGene1$backg_id.x[1]=="attP2",36303,36304))
Bk2 <- ThermDat2 %>% filter(genotype == ifelse(FGene2$backg_id.x[1]=="attP2",36303,36304))


FGene_51411_1 <- rbind(FGene1, Bk1)
FGene_51411_2 <- rbind(FGene2, Bk2)


Atpha_RNAi_plot1 <- ggplot(FGene_51411_1, aes(set_id, incapacitation, shape=set_id , color=date)) +
  geom_point(position = position_jitter(width = 0.4)) +
  ggtitle("Na pump α subunit (atpα) - 1") +
  guides(col=guide_legend("Group Mean ")) +
  ylab("Incapacitation Time (sec)") +
  xlab("RNAi Line") 



Atpha_RNAi_plot2 <- ggplot(FGene_51411_2, aes(set_id, incapacitation, shape=set_id , color=Date)) +
  geom_point(position = position_jitter(width = 0.4)) +
  ggtitle("Na pump α subunit (atpα) - 2") +
  guides(col=guide_legend("Group Mean ")) +
  ylab("Incapacitation Time (sec)") +
  xlab("RNAi Line") 

Atpha_RNAi_plot <- plot_grid(Atpha_RNAi_plot1,Atpha_RNAi_plot2,nrow=2,labels = c('a.', 'b.'), rel_widths = c(1,1.3), label_size = 10)


###Rho_RNAi_plot
foc.gene <- 35027

FGene1 <- ThermDat1 %>% filter(genotype ==foc.gene)
FGene2 <- ThermDat2 %>% filter(genotype ==foc.gene)

Bk1 <- ThermDat1 %>% filter(genotype == ifelse(FGene1$backg_id.x[1]=="attP2",36303,36304))
Bk2 <- ThermDat2 %>% filter(genotype == ifelse(FGene2$backg_id.x[1]=="attP2",36303,36304))


FGene_35027_1 <- rbind(FGene1, Bk1)
FGene_35027_2 <- rbind(FGene2, Bk2)


Rho_RNAi_plot1 <- ggplot(FGene_35027_1, aes(set_id, incapacitation, shape=set_id , color=date)) +
  geom_point(position = position_jitter(width = 0.4)) +
  ggtitle("Rho_RNAi_plot - 1") +
  #guides(col=guide_legend("Group Mean ")) +
  ylab("Incapacitation Time (sec)") +
  xlab("RNAi Line") 



Rho_RNAi_plot2 <- ggplot(FGene_35027_2, aes(set_id, incapacitation, shape=set_id , color=Date)) +
  geom_point(position = position_jitter(width = 0.4)) +
  ggtitle("Rho_RNAi_plot - 2") +
  #guides(col=guide_legend("Group Mean ")) +
  ylab("Incapacitation Time (sec)") +
  xlab("RNAi Line") 

Rho_RNAi_plot <- plot_grid(Rho_RNAi_plot1,Rho_RNAi_plot2,nrow=2,labels = c('a.', 'b.'), rel_widths = c(1,1.3), label_size = 10)



### dmr_RNAi
foc.gene <- 50656

FGene1 <- ThermDat1 %>% filter(genotype ==foc.gene)
FGene2 <- ThermDat2 %>% filter(genotype ==foc.gene)

Bk1 <- ThermDat1 %>% filter(genotype == ifelse(FGene1$backg_id.x[1]=="attP2",36303,36304))
Bk2 <- ThermDat2 %>% filter(genotype == ifelse(FGene2$backg_id.x[1]=="attP2",36303,36304))


FGene_50656_1 <- rbind(FGene1, Bk1)
FGene_50656_2 <- rbind(FGene2, Bk2)


dmr_RNAi_plot1 <- ggplot(FGene_50656_1, aes(set_id, incapacitation, shape=set_id , color=date)) +
  geom_point(position = position_jitter(width = 0.4)) +
  ggtitle("dmr_RNAi_plot - 1") +
  guides(col=guide_legend("Group Mean ")) +
  ylab("Incapacitation Time (sec)") +
  xlab("RNAi Line") 



dmr_RNAi_plot2 <- ggplot(FGene_50656_2, aes(set_id, incapacitation, shape=set_id , color=Date)) +
  geom_point(position = position_jitter(width = 0.4)) +
  ggtitle("dmr_RNAi_plot - 2") +
  #guides(col=guide_legend("Group Mean ")) +
  ylab("Incapacitation Time (sec)") +
  xlab("RNAi Line") 

dmr_RNAi_plot <- plot_grid(dmr_RNAi_plot1,dmr_RNAi_plot2,nrow=2,labels = c('a.', 'b.'), rel_widths = c(1,1.3), label_size = 10)

#merge all the plots

all_plot <- plot_grid(Rho_RNAi_plot,dmr_RNAi_plot,Atpha_RNAi_plot,nrow=3, ncol=3,labels = c('a.', 'b.', 'c.','d.'), rel_widths = c(5,5,5,5), label_size = 10)

```