---
title: "Project Notes"
author: "Patricka Williams-Simon and Elizabeth King"
date: "04/01/2023"
output: html_document
---

## Instructions for reproducing analyses

### Functions
1. `Functions/mappingfunctions.R` holds some basic functions used for QTL mapping
2. `Functions/h2_functs.R` holds functions for calculating heritability and genetic correlations with a linear model and jackknife method
3. `Functions/ggplot_theme.R` holds the plotting theme for publication format

### Obtain raw data from Zenodo

-Fetch raw data from repository (zenodo)- when done, update any relevant file paths and process raw scripts

### Phenotype analysis steps for the DSPR RILs and Founders: in the `LearnMemTolPheno` folder

1. Process raw data for learning, memory, and thermal tolerance
  - Learning and Memory
      - Processed data from HeatCalc is parsed using a perl script (`heatcalcRead_LearnMem.pl`) and outputs: `ProcessedData/HeatProc_Learn.txt`
        -use `perl myscript.pl > myoutput.txt 2> myerror.txt &` in terminal 
      - Raw `.asc` files are processed using an R script (`process_raw_learnmem.R`) to identify any other issues (see Step 2) and outputs `/ProcessedData/Learn_raw.rda`
  - Thermal tolerance
      - Raw `.asc` files are processed using an R script (`process_raw_therm.R`) to identify issues and calculate thermal tolerance for each individual and outputs `ProcessedData/Incapacitation.rda`. There are matching scripts and output for the data for the founders.

2. Raw data and processed data are checked using `check_raw_data.Rmd`. Notes within this script for inputs. Outputs the following:
  - `ThermalTol_processed.txt`
  - `LearnMem_processed.txt`
These are cleaned datasets ready for next steps. 

3. `Validate_incapacitation.R` compares human scored and automated incapacitation scores from the heatbox. Input is `ProcessedData/ThermalTol_processed.txt` and `ProcessedData/Thermotolerance_test.csv`. and Makes a plot: `Plots/HeatBox_Human_Val.pdf` and outputs the paired data: `ProcessedData/Validation.txt`.  `Compare_heatbox_heatplate.R` compares the two methods for measuring thermal tolerance (box and plate).  

4. `Processing_Pheno_data.Rmd` gets line means and does basic visualizations. Outputs: `ProcessedData/L_MDATA.rda` and `ProcessedData/T_TDATA.rda`. `Processing_ThermTol_Pheno_Founders.Rmd` does this for the founder data and outputs Figure 3b. 

5. `visualize_pheno_data.Rmd` makes several phenotype plots for the RIL data. See script for additional output plots. Outputs Figure 2 and Figure 3c. 

6. Additional script `correlation_between_traits` analyzes the relationship between thermal tolerance, learning, and memory and makes some scatterplots. These are not included in the publication. 

### QTL mapping and peak analysis: in `LearnMemTolQTL` folder

1. `h2_rg.Rmd` calculates heritabilities and genetic correlations
2. `mapping_perms` performs the QTL mapping and permutations and calculation of FDR and FWER, generating several rda files holding the results
2. `identify_peaks` finds each QTL peak and saves it in a list. `IndividualPeaks.Rmd` incorporates additional datasets to make a large summary list. There are notes within the file. The list contains our three phenotypes containing a data.frame with the following columns with one QTL per row:
  - "chr" - the chromosome where the QTL is located
  - "Ppos" - the physical position of the QTL in R5 coordinates
  - "Gpos" - the genetic position of the QTL in cM
  - "Gaxis" - the genetic position on a long scale (chromosomes stacked)
  - "LL" - the LOD score at the peak
  - "lp" - the lower bound physical position in R5 coordinates for the BCI
  - "up" - the upper bound physical position in R5 coordinates for the BCI
  - "lpchr" - the lower bound chromosome for the BCI
  - "upchr" - the upper bound chromosome for the BCI
  - "lg" - the lower bound genetic position for the BCI
  - "ug" - the uppper bound genetic position for the BCI
  - "ulod" - the LOD score at the upper bound
  - "llod" - the LOD score at the lower bound 
  - "chrR6" - the chromosome in R6 coordinates
  - "PposR6" - the physical position of the QTL in R6 coordinates
- "lpR6" the lower bound physical position in R6 coordinates for the BCI
  - "upR6" - the upper bound physical position in R6 coordinates for the BCI
  - "chrN" - the chromosome number without L & R 
"A1" - "A8" - the haplotype effect estimates (haplotype means at the QTL)
  - "A1se" - "A8se" - the standard errors of the haplotype effect estimates
  - "A1N" - "A8N" - the number of RILs with each haplotype genotype at the QTL location   
- "L1" - "L8" - the number of RILs in the "low pool" with each haplotype genotype
- "H1" - "H8" - the number of RILs in the "high pool" with each haplotype genotype 
3.  `QTL_effects_visualization_LM.Rmd` makes the plots of the haplotype effects at each QTL for learning and memory phenotypes. `TT-founder-effect.Rmd` constructs these plots for thermal tolerance data. Makes Figure 3c and Figure S4.
4. `visualize_genome_scan.Rmd` plots the genome scan and makes a composite figure - Figure 3. 
5. `individualpeaks.Rmd` plots each peak separately
6. `Visualize_Peak_Intervals.Rmd` plots each peak interval with the DE genes. Makes figures S6 and S7.
7. `ThermTol_additional_mapping.R` performs a genome scan after correcting for the large Q5 peak and one including subpopulation in the model to compare to our genome scan. Creates figure S2.

### RNAseq analysis: in `LearnMemTolRnaseq` folder

#### The alignment and assembly steps are in `LearnMemTolRnaseq/RNAseq_Processing_LearnMemTol`

1.Set_up_commands_masterpipeline.R is the master script that outputs 8 separate scripts for processing RNAseq data. It sets up scripts to run the RNA-Seq data on a cluster using SLURM. It will require adaptation for your own system and file paths.

Output sripts are named: 

 - S01_Trim_LearnMemRNA.txt: trims data
 - S02_QC_LearnMemRNA.txt: performs quality control
 - S03_Align_LearnMemRNA.txt: aligns data to the fly genome 
 - S04_SamtoBam_LearnMemRNA.txt: converts file from .sam to .bam
 - SO5_MergeBam_LearnMemRNA.txt: merges bam files
 - SO6_Assemble_LearnmemRNA.txt: assembles data

 - S08_Abundances_LearnMemRNA_eb.txt - finds abundances (Stringtie shorter version. This step only identifies genes that are already known. No novel genes are detected.)

or

 - S08_Abundances_LearnMemRNA.txt - finds abundances including novel transcripts


#### Differential Expression Analysis is in `LearnMemTolRnaseq/DEseq_LearnMemTol`

1. `prepDEseq.py` takes the output form S08_Abundances_LearnMemRNA_pl.txt, stringtie (shorter version) and converts it into a usual version for the DEseq pipeline.
2. `LearnMem_RNAproc_DESeq_eb.Rmd` and `ThermTolRNAproc_DEseq_eb.Rmd` runs the DEseq analysis pipeline on the different high and low cohorts for the RNAseq data (learning and memory and thermal tolerance respectively). Additional notes in this script. Output data is in `Data`
3. `Visualize_LMRNA_Data.Rmd` (for learning and memory) and `Visualize_TTRNA_Data.Rmd` (for thermal tolerance) create MA and volcano plots. This script creates Figure 4 and Figure S5.
4. `Visualize_ind_genes.R` creates a series of plots of the normalized counts in each group for individual genes in the Q5 interval. Creates Figure S9.


### RNAi analysis: in `LearnMemTolPheno` folder

1. The `HeatPlate_PreProcess` folder holds the python scripts for all pre processing. This includes: 
a) The script for calibrating the Raspberry Pi camera (`camera_calibration.py`).  
b) The script for collecting the image data during assays (`thermocouple_5.py`). 
c) The `fly_tracker_batch.py` script uses deeplabcut to auto track the flies from the recordings. 
d) The `Flymove.R` script processes the deeplabcut output to make a combined `.csv` file with fly positions over time for each fly. 

2. `process_raw_therm_heatplate.R` and `process_raw_therm_heatplate_val.R` processes the raw output from deeplabcut to score incapacitation for the RNAi lines and the RIL validation set. 

3. `RNAi_post_raw_process.R` checks for and corrects errors and combines the two rounds of phenotyping. 

4. `Models_RNAi.R` performs ANOVAs for each gene and the specific post-doc test for the cross vs the control cross. Outputs: `ProcessedData/RNAi_model_output.csv`

5. `RNAi_viz_individualgenes.R` makes a plot of RNAi results for every focal gene, which is Figure S8. `RNAi_viz.R` makes a plot of several genes together. 

6. `RNAi_DEseq_compare.R` collates the RNAi results and DEseq results to make Figure 5.




### Notes

A record of notes in the style of a lab notebook

### February 2023

In finalizing a table, we discovered overlap between what we called Q5 and Q6 in the BCI. Q5 was dropped, making Q6 now Q5 etc. The notes below will thus contain many refs to Q6 that are Q5 in the MS.

### December 2022 & January 2023

Went through and finalized repo and plots for MS. Check git logs from this period for notes. 

For the heat plate validation, I identified several issues with the validation set and eliminated those records. There was also a higher level of problematic tracking with pipe cleaner closure sometimes being flagged as a fly. This did not happen in the other dataset, likely due to more careful assay conditions. These records were also flagged and dropped and this can be avoided in the future with our assay checks (lighting, focus, etc)

### Jan 12, 2021

Completed new visualizaiton of RNAi results. in LearnMemTolPheno/Plots/RNAi_phenos_all.pdf -EGK

### Dec 20, 2021

Working on multiple comparisons procedures for the RNAi data. The underscore is causing issues in specifying which comparisons so we will change to symbolic names. -EGK


### April 21, 2021
Today, Libby and I started writing code to analysis the hot plate data: 
MyGitHub/LearnMemTol/LearnMemTolPheno/HeatPlate_autotrack/

I started by looking at an old script that we used to analyze ThermTol from the heatbox.




####April 12, 2021

Patricka - you were missing the ggrepel library: library(ggrepel)
Also, the separator \n was missing the n - this means new line. It just had a slash.

You had added some code adding a gname column to your TTlrt objects. This created two columns when you did the merge gname.x and gname.y. So when you try to use gname in the plot, it doesnt know what to use. I commented these out. 

Also, check that the object you read in TTlrt_condition is named right. You refer to this as TTlrt_con throughout - fine if it matches for you. It didnt for me but i may have an old rda. 

####April 12, 2021
-Libby, Can you help with adding a gradient color to my volcano plots?
-Please look at script in LearnMemTolRnaseq/DEseq_LearnMemTol/Scripts/Visualize_TTRNA_data.Rmd
-Section under volcano plot, #for different conditions: H_Incu vs L_Incu, H_RT vs L_RT, H_Incu vs H_RT, L_Incu vs L_RT
-Also, because you are here let's add the gnames to volcano plots. Genes are in object called place.g  
-Thanks!!!

###Dec. 12, 2018
-Compared the significantly deferentially expressed genes that we identified from the DEseq analysis to the gene summary on flybase. From all the genes identified, we needed to look at specific genes that may influence learning and memory based on their known functions. I took note of all the genes that were listed to play a role behavior, neurotransmitter transportation and trafficking. I also listed a few genes that are shown to interact with to a genes that was listed. Below are the list of genes: 
Learning: 
  2L: Mad, chm, Nlg2, Pvf3, Myo281
  2R: Fkbpl2, Mctp, Jheh2,DptA/B, Jabba, 5-HT1B 
  3L: MsR2, Fife, Mp
  3R: Gluclalpha, KaiR1D, Ktl, Grik, SlFaR, Nrx-1,Octbeta1R
  X: Shmt, Ca-alpha1T, Shi,Lcch3, HDAC6, CG8916
    Nlg2, Fife, Ort, Rh3

Memory:
CG1698 and Hdc


###Nov. 27 2018
-Created script to find gene (under significant qtls) summaries on fly base. Script is stored under MyGitHub/LearnMemTol/LearnMemTolQTL/Scripts/IndividualPeaks.Rmd, section Genes Summary
-Downloaded automated_gene_summaries file here
ftp://ftp.flybase.net/releases/FB2018_05/precomputed_files/genes/automated_gene_summaries.tsv.gz

### Nov 20th 2018

EGK - updated identify peaks to add haplotype estimates at each peak. Split section finding DE genes under peaks.

### Nov 9 2018 

Reorganized Pheno and QTL folders. PWS will do same for DEseq

#Nov. 07th, 2018
-continued working on identifying which haplotype is positioned at significant qtls.

#Oct. 28th, 2018
-Identify genesfounder haplotype that are in low and high cohorts. File is named identify_founder_additive.Rmd
-Started working on script for individual QTL plots to show where all the significanly DE genes are located under the peaks.
-Some padj values are set to NA. This happens when the count is extreme (low or high). See this link and add it to your project notes so we dont forget why this is happening. https://support.bioconductor.org/p/76144/

#October 19, 2018
Reorganize files. May need to check paths etc. for many scripts.
Should we set up a script to fetch the raw data from zenodo (or wherever)

#Sep. 30, 2018

Issues with the gene map file
  - The gene map table being used is old (from 2015). I got the new version. 
  - The file was zipped so I unzipped it
  - There are other Drosophila species included so we need to subset to D. mel
  
Updated peaks list. We need to map thermal tolerance after correcting for the large peak & recalculate fdr.

Added code to look at differentially expressed genes within a peak interval. Needs to be applied to each peak.


#Sep. 20, 2018

- Added code to split gene positions to identify.peaks script. Also added a couple notes for checking some things. -EGK

#Sep. 19, 2018

-Today I tried to identfy the genes that are located under the qtl peaks. The first attempt was not successful, because I need to split a col in the dataset (gene_map_table), but with no avail. Below is an example:

2L:8466817..8488671(1), I need to remove :, .., and (1), then split the rest of the info into separate columns. I will try again tonight. (PW-S)


#Sep. 14, 2018

- Updated mapping scripts to calculate CIs and convert coordinates to release 6.

#Sep 06, 2018 (moved notes from script to here)

- Generate a list of gtf commands - ref p.92-93 Dunn book

  -grep -li assemb *.gtf > ~/LearnMemTolRnaseq/DEseq_LearnMemTol/Data/gtf_list.txt
  -searches and list all .gtf which includes assemb*

-Open gtf_list.txt in TextWrangler [command+F]
-In the search window type ^
  -In the replace window, type full path i.e: ~/LearnMemTolRnaseq/DEseq_LearnMemTol/Scripts/
  -Do Replace All
  -This gives full path to each of the .gtf

Alternatively:
  -While in processed/ (i.e. where directory "ballgown" is located), run:
  -python ../LearnMemTolRnaseq/DEseq_LearnMemTol/Scripts/prepDE.py
  -Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`


#Sep 03, 2018
-Updated script LearnMem_RNAproc_DESEq_eb. Ran the data through the DESeq pipeline, however when i got to the step where I make the SVs a part of the dataset i got a error.

code: LMdat.sva$SV1 <- LMsvobj$sv[,1]
error:Error in `$<-.data.frame`(`*tmp*`, SV1, value = c(-0.358398470470704,  : 
  replacement has 20 rows, data has 16994

solution: I was using the sva dataset , whereas i was suppose to be using the TTpheno dataset (silly Patricka :))

_When running DESeq with SV in the model there was 3 rows that did not converge in beta.



#Aug 27, 2018

-Decided to try the the 'shorter version' of the tuxedo pipeline for rna seq processing. ref:http://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#de.
-This simplified workflow attempts to directly estimate and analyze the expression of a known set of transcripts as given in the reference annotation file.
-One caveat to this work flow is that novel transcripts will not be identified.
(PW-S)


#Aug 24, 2018
- Worked on the ThermTolRNAproc_DESeq script today. There are different MSTRG with the same FBgn (FBgn0002781). I tried giving every FBgn0002781 a different identify, but with no avail, since there are several rows with the same FBgn0002781. (pw-s)   

#Aug 23, 2018
-Ran the stringtie code again in Lewis, in attampt to get more genes idifified (converted from MSTRG #'s to FBgn's). Several were converted. 



#July 26, 2018
-created script for Rna-seq processing using DEseq (RNAproc_DEseq.Rmd)

(PW-S)

#July 23, 2018

-updated script to visualize genome scans - qtl maps - in base R.
-made qtl map with all 3 traits
(PW-S)


### July 22, 2018

- fixed some errors in the permutation script
- FDR is lower than expected. Check observed peaks to confirm we are not double counting peaks.
- FWER is as expected

### July 20, 2018

-ran genome scans
-ran permutation tests
-updated vizualize_genome_scan
(PW-S)


- Made a script for plotting the genome scan (vizualize_genome_scan). 
- Added code to perform the permutations. 
- Also renamed loglikmulti to mapping functions as it now holds a couple functions 
(EGK)

###July 17, 2018
-updated processing_pheno_data.rmd. The phenotypic data is ready to move onto the next stage, genome scans.
-updated, mapping_perrms.R. Libby wrote this script and i ran the codes today. The idea is to run all three traits in one genome scan and one permutaiton test with all three traits to control for errors.


###July 16, 2018
-wrote scripts for analysing phenodata (MyGitHub/LearnMemTol/Scripts/Processing_Pheno_data.Rmd). Learning does not have a normal distribution, therefore we need to transform the data. I tried several transformation techniques (log, log10, square root and arcsign),but none worked. Libby suggested doing a quantile normalization.

-wrote scripts for analysing correlation between traits (MyGitHub/LearnMemTol/Scripts/correlation_between_traits.Rmd).


###July 11, 2018

-Learned Git processing via Nivalis
-To embed Git into Nivalis, fellow dierections below

  git config --global user.name "YourUserName"
  git config --global user.email "YourName@example.com"

  git clone https://YourUserName@github.com/OwnersUserName/repository.git

  git clone https://egking@github.com/egking/HaplotypeEstimator.git
  
-Practicing git 


### July 9

- Finalize dataset for mapping. We decided as a group to eliminate flies that were incapacitated on the punishment side and never resumed moving after that. See check_raw_data.Rmd for notes.

### June 18 
- Starting git and this record. Old notes will be moved over. 
- Checked raw processing scripts. 
- Checked validation script for incapacitation. 

###March 04, 2020
-loaded raw founder heatbox dataset
-processed the raw data


### Sep 10, 2020
-identified haplotype effect at Q6 ( file saved: LearnMemTol/LearnMemTolQTL/Scripts/TT-founder-effect-Q6.Rmd)



