---
title: "Project Notes"
author: "Patricka Williams-Simon and Elizabeth King"
date: "4/19/2018"
output: html_document
---

# Table of Contents

## To do


## Analysis Workflow 

1. Process raw data for learning, memory, and thermal tolerance
  - Learning and Memory
      - Processed data from HeatCalc is parsed using a perl script (heatcalcRead_LearnMem.pl) and outputs: HeatProc_Learn.txt
      - Raw .asc files are processed using an R script (process_raw_learnmem.R) to identify any other issues (see Step 2)
  - Thermal tolerance
      - Raw .asc files are processed using an R script (process_raw_therm.R) to identify issues and calculate thermal tolerance for each individual
      - Validation of thermotolerance method is validate_incapacitation.R      

2. Raw data and processed data are checked using check_raw_data.Rmd. Notes within this script.

## Notes

A record of notes in the style of a lab notebook

#Sep 03, 2018
-Updated script LearnMem_RNAproc_DESEq_eb. Ran the data through the DESeq pipeline, however when i got to the step where I make the SVs a part of the dataset i got a error.

code: LMdat.sva$SV1 <- LMsvobj$sv[,1]
error:Error in `$<-.data.frame`(`*tmp*`, SV1, value = c(-0.358398470470704,  : 
  replacement has 20 rows, data has 16994





#Aug 27, 2018

-Decided to try the the 'shorter version' of the tuxedo pipeline for rna seq processing. ref:http://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#de.
-This simplified workflow attempts to directly estimate and analyze the expression of a known set of transcripts as given in the reference annotation file.
-One caveat to this work flow is that novel transcripts will not be identified.
(PW-S)


#Aug 24, 2018
- Worked on the ThermTolRNAproc_DESeq script today. There are different MSTRG with the same FBgn (FBgn0002781). I tried giving every FBgn0002781 a different identify, but with no avail, since there are several rows with the same FBgn0002781. (pw-s)   

#Aug 23, 2018
-Ran the stringtie code again in Lewis, in attampt to get more genes idifified (converted from MSTRG #'s to FBgn's). Several were converted. 



#July 26, 2018
-created script for Rna-seq processing using DEseq (RNAproc_DEseq.Rmd)

(PW-S)

#July 23, 2018

-updated script to visualize genome scans - qtl maps - in base R.
-made qtl map with all 3 traits
(PW-S)


### July 22, 2018

- fixed some errors in the permutation script
- FDR is lower than expected. Check observed peaks to confirm we are not double counting peaks.
- FWER is as expected

### July 20, 2018

-ran genome scans
-ran permutation tests
-updated vizualize_genome_scan
(PW-S)


- Made a script for plotting the genome scan (vizualize_genome_scan). 
- Added code to perform the permutations. 
- Also renamed loglikmulti to mapping functions as it now holds a couple functions 
(EGK)

###July 17, 2018
-updated processing_pheno_data.rmd. The phenotypic data is ready to move onto the next stage, genome scans.
-updated, mapping_perrms.R. Libby wrote this script and i ran the codes today. The idea is to run all three traits in one genome scan and one permutaiton test with all three traits to control for errors.


###July 16, 2018
-wrote scripts for analysing phenodata (MyGitHub/LearnMemTol/Scripts/Processing_Pheno_data.Rmd). Learning does not have a normal distribution, therefore we need to transform the data. I tried several transformation techniques (log, log10, square root and arcsign),but none worked. Libby suggested doing a quantile normalization.

-wrote scripts for analysing correlation between traits (MyGitHub/LearnMemTol/Scripts/correlation_between_traits.Rmd).


###July 11, 2018

-Learned Git processing via Nivalis
-To embed Git into Nivalis, fellow dierections below

  git config --global user.name "YourUserName"
  git config --global user.email "YourName@example.com"

  git clone https://YourUserName@github.com/OwnersUserName/repository.git

  git clone https://egking@github.com/egking/HaplotypeEstimator.git
  
-Practicing git 


### July 9

- Finalize dataset for mapping. We decided as a group to eliminate flies that were incapacitated on the punishment side and never resumed moving after that. See check_raw_data.Rmd for notes.

### June 18 

- Starting git and this record. Old notes will be moved over. 
- Checked raw processing scripts. 
- Checked validation script for incapacitation. 


