---
title: "Project Notes"
author: "Patricka Williams-Simon and Elizabeth King"
date: "4/19/2018"
output: html_document
---

# Table of Contents

## To do


## Analysis Workflow 

-Fetch data from repository (zenodo)- when done, update .pl and process raw scripts

### Functions
1. mappingfunctions.R () holds some basic functions used for QTL mapping
2. ggplot_theme holds the plotting theme for publication format

### Phenotype steps are in LearnMemTolPheno folder
1. Process raw data for learning, memory, and thermal tolerance
  - Learning and Memory
      - Processed data from HeatCalc is parsed using a perl script (heatcalcRead_LearnMem.pl) and outputs: HeatProc_Learn.txt
      - Raw .asc files are processed using an R script (process_raw_learnmem.R) to identify any other issues (see Step 2)
  - Thermal tolerance
      - Raw .asc files are processed using an R script (process_raw_therm.R) to identify issues and calculate thermal tolerance for each individual
      - Validation of thermotolerance method is validate_incapacitation.R      

2. Raw data and processed data are checked using check_raw_data.Rmd. Notes within this script.

3. Processing_Pheno_data get line means etc.
4. Visualize_pheno_data makes phenotype plots

### QTL mapping is in LearnMemTolQTL

1. mapping_perms performs the QTL mapping and permutations and calculation of FDR and FWER
2. identify_peaks finds each QTL peak and saves it in a list.
3. genes_under_peaks.R  looks at the DE genes from RNAseq (see below) to see which DE genes are under each peak
4. visualize_genome_scan.Rmd plots the genome scan
5. GeneSums.Rmd looks at FlyBase summaries for the genes in each interval
6. individualpeaks.Rmd plots each peak separately
7. Visualize_Peak_Intervals.Rmd plots each peak interval with the DE genes


### Scripts to process the RNAseq data was first created on Nivalis (lab server).
1.Set_up_commands_masterpipeline.R is the master script that outputs 8 separate scripts for processing RNAseq data. both he mastr scripts and the outputs are saved under home/DSPR/Scripts/rna-seq_processing_data. 

Output sripts are named: 
S01_Trim_LearnMemRNA.txt - trims data
S02_QC_LearnMemRNA.txt - performs quality control
S03_Align_LearnMemRNA.txt - aligns data to the fly genome 
S04_SamtoBam_LearnMemRNA.txt - converts file from .sam to .bam
SO5_MergeBam_LearnMemRNA.txt - merges bam files
SO6_Assemble_LearnmemRNA.txt - assembles data
#S07_CompareRef_LearnMemRNA.txt - compares data to reference genome 
S08_Abundances_LearnMemRNA.txt - finds abundances
####
S08_Abundances_LearnMemRNA_pl.txt - creates a list of abundances to be used by Perl. 
S08_Abundances_LearnMemRNA_eb.txt - finds abundances (Stringtie shorter version. This step only identifies genes that are already known. No novel genes are detected.)
S08_Abundances_LearnMemRNA.txt - finds abundances including novel transcripts


N.B. All output scripts were copied to Lewis where the RNAseq analysis was processed. 

###Raw data (from Rapid Genomes) were transfered into a file list and is saved /home/pwilliams/DSPR/Scripts/rna-seq_processing_data/filelist.rda



### RNA-seq Analysis in LearnMemTolQTL, within this folder there is a trim and quality control folder, the tuxedo pipeline analysis floder, and the differential expression folder.

### Trim and quality control (this step of the Rna-seq ananlysis was processed in Lewis.) /home/MyGitHub/LearnMemTol/LearnMemTolRnaseq/Trim_QC_LearnMemTol/Scripts.

## Trimming (ref:https://cutadapt.readthedocs.io/en/stable/guide.html )
1. S01_Trim_LearnMemRNA.txt - trims data. File is stored on Lewis under "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_raw/S01_Trim_LearnMemRNA.txt/"
2. S01_sbatch_trimRNA.sh - sbatch file that runs job array on S01_Trim_LearnMemRNA.txt.   "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_raw/S01_sbatch_trimRNA.sh". To run sbatch file on the Lewis terminal use, sbatch --array=1-228 sbatch_trimRNA.sh
3. Generated .fastqc files are outputted into  "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_trimmed/"

## Qulaity Control (ref: https://dnacore.missouri.edu/PDF/FastQC_Manual.pdf)
1. S02_QC_LearnMemRNA.txt - performs quality control on trimmed files. File is stored "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_trimmed/"
2. S02_sbacth_QC.sh - runs job array on S02_QC_LearnMemRNA.txt "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_trimmed/S02_sbatch_QC.sh". To run sbatch file on the Lewis terminal use, sbatch --array=1-228 sbatch_QC.sh
3. Generated .fastq.gz files are outputted into "/storage/hpc/group/kinglab/Patricka/Learn_Mem/fastqc_files/"

###TuxPipeline_LearnMemTol (ref: https://www.nature.com/articles/nprot.2016.095)

##HISAT - Aligns transcripts to fly genome
1. S03_Align_LearnMemRNA.txt. File is stored "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_trimmed/sbatch_Align_LearnMem.sh"
2. S03_sbatch_Align_LearnMem.sh - runs job array on S03_Align_LearnMemRNA.txt. file is stored "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_trimmed/S03_sbatch_Align_LearnMem.sh".To run sbatch file on Lewis use, sbatch --array=1-228 sbatch_Align_LearnMem.sh
3. Generated .sam files are outputted into "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_aligned/"

##Samtools sort  - sorts and converts .sam files into .bam, so that they are usable by                stringtie.
1. S04_SamtoBam_LearnMemRNA.txt. File is stored "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_aligned/S04_SamToBam_LearnMemRNA.txt"
2. S04_sbatch_convertsambamRNA.sh, runs a job array on S04_SamtoBam_LearnMemRNA.txt. File is stored on lewis,  "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_aligned/S04_sbatch_convertsambamRNA.sh"
3.Generated .bam files are saved to "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_bam/"


## Samtools merge - merges bam files.
1. S05_MergeBam_LearnMemRNA.txt. File is saved "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_bam/S05_MergeBam_LearnMemRNA.txt"
2. S05_sbatch_mergebamRNA.sh, runs job array on S05_MergeBam_LearnMemRNA.txt. File is saved "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_bam/S05_sbatch_mergebamRNA.sh". To run sbatch file on Lewis use, sbatch --array=1-114 sbatch_mergebamRNA.sh
3. Geberated files are saved to "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_bam/"

## Stringtie - assembles transcripts to genome.
1. SO6_Assemble_LearnmemRNA.txt. File is saved "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_bam/S06_Assemble_LearnMemRNA.txt"
2. S06_sbatch_assembleRNA.sh runs job array on SO6_Assemble_LearnmemRNA.txt "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_bam/S06_sbatch_assembleRNA.sh"
3. Generated files are outputted into "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_assemble/"

##S07_CompareRef_LearnMemRNA.txt - compares data to reference genome

## Stringtie (shorter version) - finds abundances
1. S08_Abundances_LearnMemRNA_eb.txt "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_stringtie_eb/S08_Abundances_LearnMemRNA_eb.txt"
2. S08_sbatch_abundancesRNA_eb.sh runs job array on S08_Abundances_LearnMemRNA_eb.txt "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_stringtie_eb/S08_sbatch_abundancesRNA_eb.sh"
3. Generated files are saved here "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_stringtie_eb/rna-seq_ballgown_eb/"
4.S08_Abundances_LearnMemRNA_pl.txt - creates a list of abundances to be used by Perl.

### after this step, in the shorter version of stringtie pipeline, the data is exported to the DEseq pipeline to continue analysis using DEseq count data. 


#####still working on this section
## Stringtie (longer version) - finds abundances
1.S08_Abundances_LearnMemRNA.txt "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_stringtie_eb/S08_Abundances_LearnMemRNA.txt"
2, S08_sbatch_abundancesRNA.sh runs job array on S08_Abundances_LearnMemRNA.txt "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_stringtie_eb/sbatch_abundancesRNA.sh"
3. Generated files are saved here "/storage/hpc/group/kinglab/Patricka/Learn_Mem/rna-seq_data_ballgown/" 


### DEseq_LearnMemTol/Scripts

1. prepDEseq.py takes the output form S08_Abundances_LearnMemRNA_pl.txt, stringtie (shorter version) and converts it into a usual version for the DEseq pipeline.
2. LearnMem_RNAproc_DESeq_eb.Rmd runs the DEseq analysis pipeline on the learning and memory RNAseq data.
3. ThermTolRNAproc_DESeq_eb.Rmd runs the DEseq analysis pipeline on the thermal tolerance RNAseq data.
4. Visualize_LMRNA_Data.Rmd and Visualize_TTRNA_Data.Rmd both plot graphs for learning, memorry, and thermal tolerance.


###Ballgown - identifies Defrentially expressed genes for the longer version of the stringtie. This step in the analysis was done on Navalis (lab server) /home/pwilliams/DSPR/Scripts/rna-seq_processing_data/ballgown_processing.R 



### Notes

A record of notes in the style of a lab notebook

###Dec. 12, 2018
-Compared the significantly defferentially expressed genes that we identified from the DEseq analysis to the gene summary on flybase. From all the genes identified, we needed to look at specific genes that may influence learning and memory based on their known functions. I took note of all the genes that were listed to play a role behavior, neurotransmitter transportation and trafficking. I also listed a few genes that are shown to interact with to a genes that was listed. Below are the list of genes: 
Learning: 
  2L: Mad, chm, Nlg2, Pvf3, Myo281
  2R: Fkbpl2, Mctp, Jheh2,DptA/B, Jabba, 5-HT1B 
  3L: MsR2, Fife, Mp
  3R: Gluclalpha, KaiR1D, Ktl, Grik, SlFaR, Nrx-1,Octbeta1R
  X: Shmt, Ca-alpha1T, Shi,Lcch3, HDAC6, CG8916
    Nlg2, Fife, Ort, Rh3

Memory:
CG1698 and Hdc


###Nov. 27 2018
-Created script to find gene (under significant qtls) summaries on fly base. Script is stored under MyGitHub/LearnMemTol/LearnMemTolQTL/Scripts/IndividualPeaks.Rmd, section Genes Summary
-Downloaded automated_gene_summaries file here
ftp://ftp.flybase.net/releases/FB2018_05/precomputed_files/genes/automated_gene_summaries.tsv.gz

### Nov 20th 2018

EGK - updated identify peaks to add haplotype estimates at each peak. Split section finding DE genes under peaks.

### Nov 9 2018 

Reorganized Pheno and QTL folders. PWS will do same for DEseq

#Nov. 07th, 2018
-continued working on identifying which haplotype is positioned at significant qtls.

#Oct. 28th, 2018
-Identify genesfounder haplotype that are in low and high cohorts. File is named identify_founder_additive.Rmd
-Started working on script for individual QTL plots to show where all the significanly DE genes are located under the peaks.
-Some padj values are set to NA. This happens when the count is extreme (low or high). See this link and add it to your project notes so we dont forget why this is happening. https://support.bioconductor.org/p/76144/

#October 19, 2018
Reorganize files. May need to check paths etc. for many scripts.
Should we set up a script to fetch the raw data from zenodo (or wherever)

#Sep. 30, 2018

Issues with the gene map file
  - The gene map table being used is old (from 2015). I got the new version. 
  - The file was zipped so I unzipped it
  - There are other Drosophila species included so we need to subset to D. mel
  
Updated peaks list. We need to map thermal tolerance after correcting for the large peak & recalculate fdr.

Added code to look at differentially expressed genes within a peak interval. Needs to be applied to each peak.


#Sep. 20, 2018

- Added code to split gene positions to identify.peaks script. Also added a couple notes for checking some things. -EGK

#Sep. 19, 2018

-Today I tried to identfy the genes that are located under the qtl peaks. The first attempt was not successful, because I need to split a col in the dataset (gene_map_table), but with no avail. Below is an example:

2L:8466817..8488671(1), I need to remove :, .., and (1), then split the rest of the info into separate columns. I will try again tonight. (PW-S)


#Sep. 14, 2018

- Updated mapping scripts to calculate CIs and convert coordinates to release 6.

#Sep 06, 2018 (moved notes from script to here)

- Generate a list of gtf commands - ref p.92-93 Dunn book

  -grep -li assemb *.gtf > ~/LearnMemTolRnaseq/DEseq_LearnMemTol/Data/gtf_list.txt
  -searches and list all .gtf which includes assemb*

-Open gtf_list.txt in TextWrangler [command+F]
-In the search window type ^
  -In the replace window, type full path i.e: ~/LearnMemTolRnaseq/DEseq_LearnMemTol/Scripts/
  -Do Replace All
  -This gives full path to each of the .gtf

Alternatively:
  -While in processed/ (i.e. where directory "ballgown" is located), run:
  -python ../LearnMemTolRnaseq/DEseq_LearnMemTol/Scripts/prepDE.py
  -Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`


#Sep 03, 2018
-Updated script LearnMem_RNAproc_DESEq_eb. Ran the data through the DESeq pipeline, however when i got to the step where I make the SVs a part of the dataset i got a error.

code: LMdat.sva$SV1 <- LMsvobj$sv[,1]
error:Error in `$<-.data.frame`(`*tmp*`, SV1, value = c(-0.358398470470704,  : 
  replacement has 20 rows, data has 16994

solution: I was using the sva dataset , whereas i was suppose to be using the TTpheno dataset (silly Patricka :))

_When running DESeq with SV in the model there was 3 rows that did not converge in beta.



#Aug 27, 2018

-Decided to try the the 'shorter version' of the tuxedo pipeline for rna seq processing. ref:http://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#de.
-This simplified workflow attempts to directly estimate and analyze the expression of a known set of transcripts as given in the reference annotation file.
-One caveat to this work flow is that novel transcripts will not be identified.
(PW-S)


#Aug 24, 2018
- Worked on the ThermTolRNAproc_DESeq script today. There are different MSTRG with the same FBgn (FBgn0002781). I tried giving every FBgn0002781 a different identify, but with no avail, since there are several rows with the same FBgn0002781. (pw-s)   

#Aug 23, 2018
-Ran the stringtie code again in Lewis, in attampt to get more genes idifified (converted from MSTRG #'s to FBgn's). Several were converted. 



#July 26, 2018
-created script for Rna-seq processing using DEseq (RNAproc_DEseq.Rmd)

(PW-S)

#July 23, 2018

-updated script to visualize genome scans - qtl maps - in base R.
-made qtl map with all 3 traits
(PW-S)


### July 22, 2018

- fixed some errors in the permutation script
- FDR is lower than expected. Check observed peaks to confirm we are not double counting peaks.
- FWER is as expected

### July 20, 2018

-ran genome scans
-ran permutation tests
-updated vizualize_genome_scan
(PW-S)


- Made a script for plotting the genome scan (vizualize_genome_scan). 
- Added code to perform the permutations. 
- Also renamed loglikmulti to mapping functions as it now holds a couple functions 
(EGK)

###July 17, 2018
-updated processing_pheno_data.rmd. The phenotypic data is ready to move onto the next stage, genome scans.
-updated, mapping_perrms.R. Libby wrote this script and i ran the codes today. The idea is to run all three traits in one genome scan and one permutaiton test with all three traits to control for errors.


###July 16, 2018
-wrote scripts for analysing phenodata (MyGitHub/LearnMemTol/Scripts/Processing_Pheno_data.Rmd). Learning does not have a normal distribution, therefore we need to transform the data. I tried several transformation techniques (log, log10, square root and arcsign),but none worked. Libby suggested doing a quantile normalization.

-wrote scripts for analysing correlation between traits (MyGitHub/LearnMemTol/Scripts/correlation_between_traits.Rmd).


###July 11, 2018

-Learned Git processing via Nivalis
-To embed Git into Nivalis, fellow dierections below

  git config --global user.name "YourUserName"
  git config --global user.email "YourName@example.com"

  git clone https://YourUserName@github.com/OwnersUserName/repository.git

  git clone https://egking@github.com/egking/HaplotypeEstimator.git
  
-Practicing git 


### July 9

- Finalize dataset for mapping. We decided as a group to eliminate flies that were incapacitated on the punishment side and never resumed moving after that. See check_raw_data.Rmd for notes.

### June 18 
- Starting git and this record. Old notes will be moved over. 
- Checked raw processing scripts. 
- Checked validation script for incapacitation. 

###March 04, 2020
-loaded raw founder heatbox dataset
-processed the raw data


###March 23, 2020
- 


