---
title: "Finding Haplotype Positions"
author: "Patricka Williams-Simon"
date: "10/24/2018"
output: html_document
---

```{r setup, include=FALSE}
library(DSPRqtlDataA)
library(DSPRqtl)
library(tidyverse)
library(cowplot)
source(file="../../Functions/ggplot_theme.R")

```

For each peak:
 - identify haplotype means
 - identify eQTL under peaks
 - identify eQTL that show means like haploype means
 - identify eQTL that mirror effect of DEgenes
 - idenitfy DE groups by haplotype

# Get expression data and check for highly correlated genes

```{r}

load(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/Final_EXPR_MAT.rda")

load(file="../../LearnMemTolPheno/ProcessedData/L_MDATA.rda")
ww<- which(L_MDATA$patRIL %in% rownames(qn.mat))
wwe<- which(rownames(qn.mat) %in% L_MDATA$patRIL)

L.sub <- L_MDATA[ww,]
qn.mat <- qn.mat[wwe,]

all.equal(as.numeric(row.names(qn.mat)), L.sub$patRIL)

ccs.l <- cor(L.sub$Learning_Mean, qn.mat)
ccs.m <- cor(L.sub$Memory_Mean, qn.mat)

cors.lm <- data.frame('ID' = colnames(qn.mat), 'rLearn' = as.numeric(ccs.l), 'rMem' = as.numeric(ccs.m), stringsAsFactors=FALSE)


cgn<-read.table(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/fbgn_fbtr_fbpp_fb_2012_02.tsv",header=FALSE,stringsAsFactors=FALSE,sep="\t",fill=TRUE)
load(file='/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/probes/CDS/completeCDS.rda')
ss<-merge(cgn,bl, by.x="V2",by.y="FBtr")
ww.t <- which(ss$CGid %in% colnames(qn.mat))
ww.g <- which(ss$CGgn %in% colnames(qn.mat))
decode.n <- ss[,c('V1','CGid','CGgn')]
decode.n$ID <- NA
decode.n$ID[ww.t]<- ss$CGid[ww.t]
decode.n$ID[ww.g]<- ss$CGgn[ww.g]
decode.n <- decode.n[,c('V1','ID')]
decode.n <- unique(decode.n, MARGIN=1)
colnames(decode.n) <- c("FBgn","ID")

cors.lm <- merge(cors.lm, decode.n,by='ID',sort=FALSE, all.x=TRUE)

```

#look at haps of pools - add to list

```{r}
load(file ="../ProcessedData/Peaks_wCIs.rda")

rrs <- read.table(file="../../LearnMemTolRnaseq/DEseq_LearnMemTol/Data/RILs_in_pools.txt", sep="\t", stringsAsFactors = FALSE, header=TRUE)

for(kk in 1:3)
{

list.peak <- ci.peak[[kk]]
list.peak[,43:58]<-NA
colnames(list.peak)[43:58] <- c("L1","L2","L3","L4","L5","L6","L7","L8",
                              "H1","H2","H3","H4","H5","H6","H7","H8")

for(jj in 1:nrow(list.peak))
{
  ppeak <- list.peak[jj,]
  
  objname<-paste("A_",ppeak$chr,"_",ppeak$Ppos,sep="")
  data(list=objname)
  geno <- get(objname)
  geno <- geno[,c("ril", "AA1","AA2","AA3","AA4","AA5","AA6","AA7","AA8")]
  lowset <- geno %>% filter(ril %in% rrs[rrs$Pool=='LL','patRIL'])
  highset <- geno %>% filter(ril %in% rrs[rrs$Pool=='LH','patRIL'])
  ppeak[,43:50]<-round(colSums(lowset[,2:9]))
  ppeak[,51:58]<-round(colSums(highset[,2:9]))
  list.peak[jj,]<-ppeak
}
ci.peak[[kk]]<-list.peak
}

```


# For each peak, get means and compare to eQTL means. Also compare to genes in intervals from cohort expression expt.

```{r}
data("positionlist_wgenetic")

#files from old proj
load(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/goodpeaksinfo_matrix.rda")
load(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/means_wcorrected.rda")

effects <- merge(effects, decode.n, by='ID',sort=FALSE, all.x=TRUE)

load(file="../ProcessedData/Learn_genes_under_peak.rda")

for(jj in 1:length(learn.list))
{

ll.foc <- learn.list[[jj]]
pp.foc <- ci.peak[[1]][jj,]

mw <- (which(pinfo$chr==pp.foc$chr & 
                 ((pinfo$peaklpB <= pp.foc$up & pinfo$peakupB >= pp.foc$lp))))  
  
ee <- effects[effects$ID %in% pinfo$ID[mw],]

f.high <- which(apply(rbind(as.numeric(pp.foc[,35:42]), as.matrix(ee[,34:41])),2, min) > 4)

qtl.e <- pp.foc[,19:26]
eqtl.e <- ee[,2:9]

qtl.e <- qtl.e[,f.high]
eqtl.e <- eqtl.e[,f.high]

cors.effs <- data.frame('FBgn'= ee$FBgn, 'corEff'= apply(eqtl.e, 1, function(x) cor(x,as.numeric(qtl.e))), stringsAsFactors = FALSE)

ll.all <- merge(ll.foc, cors.effs, by='FBgn', all=TRUE)

ll.all <- merge(ll.all, cors.lm, by='FBgn', all.x=TRUE)
learn.list[[jj]] <- ll.all
}


load(file="../ProcessedData/Mem_genes_under_peak.rda")

for(jj in 1:length(Mem.list))
{

ll.foc <- Mem.list[[jj]]
pp.foc <- ci.peak[[1]][jj,]

mw <- (which(pinfo$chr==pp.foc$chr & 
                 ((pinfo$peaklpB <= pp.foc$up & pinfo$peakupB >= pp.foc$lp))))  
  
ee <- effects[effects$ID %in% pinfo$ID[mw],]

f.high <- which(apply(rbind(as.numeric(pp.foc[,35:42]), as.matrix(ee[,34:41])),2, min) > 4)

qtl.e <- pp.foc[,19:26]
eqtl.e <- ee[,2:9]

qtl.e <- qtl.e[,f.high]
eqtl.e <- eqtl.e[,f.high]

cors.effs <- data.frame('FBgn'= ee$FBgn, 'corEff'= apply(eqtl.e, 1, function(x) cor(x,as.numeric(qtl.e))), stringsAsFactors = FALSE)

ll.all <- merge(ll.foc, cors.effs, by='FBgn', all=TRUE)

ll.all <- merge(ll.all, cors.lm, by='FBgn', all.x=TRUE)
Mem.list[[jj]] <- ll.all
}

```

## Figure out candidate genes

```{r}
zz<-10

pp.set <- learn.list[[zz]]
pp.set[which.max(abs(pp.set$log2FoldChange)),]
pp.set[which.min(pp.set$padj),]
pp.set[which.max(abs(pp.set$corEff)),]

pp.set[which(is.na(pp.set$log2FoldChange)==FALSE & is.na(pp.set$corEff)==FALSE),]

ci.peak.set <- ci.peak[[1]][zz,]

pp.set.s <- pp.set[is.na(pp.set$log2FoldChange)==FALSE,]
pp.set.s$cisyn <- 'Y'
pp.set.s$cisyn[is.na(pp.set.s$corEff)] <- 'N'
pp.set.s <- unique(pp.set.s[,c(1,2,4,5,6,7,8,13)], MARGIN=1)
pp.set.s$Mstartp <- pp.set.s$startp/1e6

pbuffer <- 0.04
  
pgene <- ggplot(pp.set.s, aes(Mstartp, log2FoldChange, color=cisyn, label=gname))+
  geom_point()+
  geom_text_repel(show.legend = FALSE) +
  xlim(c(ci.peak.set$lpR6/1e6 - pbuffer, ci.peak.set$upR6/1e6 + pbuffer)) +
  xlab("Position (Mbp)")+
  ylab(expression("-log"[2]*"(Fold Change)")) +
  scale_color_manual(values=c('black','red2'), name="cis eQTL") +
  my_theme
  

pgene

```

show all genes
revamp peak info
pay attention to r5 or r6


Location
Fold Change
P value

Low High pools

effects



