---
title: "Identifying Genes Under Peaks"
author: "Patricka Williams-Simon"
date: "10/24/2018"
output: html_document
---

```{r setup, include=FALSE}
library(DSPRqtlDataA)
library(DSPRqtl)
library(tidyverse)
library(DESeq2)

```

For each peak:
 - identify haplotype means
 - identify eQTL under peaks
 - identify eQTL that show means like haploype means
 - identify eQTL that mirror effect of DEgenes
 - idenitfy DE groups by haplotype

# Get expression data and check for highly correlated genes

```{r}

load(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/Final_EXPR_MAT.rda")

load(file="../../LearnMemTolPheno/ProcessedData/L_MDATA.rda")
ww<- which(L_MDATA$patRIL %in% rownames(qn.mat))
wwe<- which(rownames(qn.mat) %in% L_MDATA$patRIL)

L.sub <- L_MDATA[ww,]
qn.mat <- qn.mat[wwe,]

all.equal(as.numeric(row.names(qn.mat)), L.sub$patRIL)

ccs.l <- cor(L.sub$Learning_Mean, qn.mat)
ccs.m <- cor(L.sub$Memory_Mean, qn.mat)

cors.lm <- data.frame('ID' = colnames(qn.mat), 'rLearn' = as.numeric(ccs.l), 'rMem' = as.numeric(ccs.m), stringsAsFactors=FALSE)


cgn<-read.table(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/fbgn_fbtr_fbpp_fb_2012_02.tsv",header=FALSE,stringsAsFactors=FALSE,sep="\t",fill=TRUE)
load(file='/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/probes/CDS/completeCDS.rda')
ss<-merge(cgn,bl, by.x="V2",by.y="FBtr")
ww.t <- which(ss$CGid %in% colnames(qn.mat))
ww.g <- which(ss$CGgn %in% colnames(qn.mat))
decode.n <- ss[,c('V1','CGid','CGgn')]
decode.n$ID <- NA
decode.n$ID[ww.t]<- ss$CGid[ww.t]
decode.n$ID[ww.g]<- ss$CGgn[ww.g]
decode.n <- decode.n[,c('V1','ID')]
decode.n <- unique(decode.n, MARGIN=1)
colnames(decode.n) <- c("FBgn","ID")

cors.lm <- merge(cors.lm, decode.n,by='ID',sort=FALSE, all.x=TRUE)

```

#look at haps of pools - add to list

```{r}
load(file ="../ProcessedData/Peaks_wCIs.rda")

rrs <- read.table(file="../../LearnMemTolRnaseq/DEseq_LearnMemTol/Data/RILs_in_pools.txt", sep="\t", stringsAsFactors = FALSE, header=TRUE)

rrcode <- c('LH','LL',
  'MH','ML',
  'TH','TL')
cc <- 1

for(kk in 1:3)
{

list.peak <- ci.peak[[kk]]
list.peak[,43:58]<-NA
colnames(list.peak)[43:58] <- c("L1","L2","L3","L4","L5","L6","L7","L8",
                              "H1","H2","H3","H4","H5","H6","H7","H8")
for(jj in 1:nrow(list.peak))
{
  ppeak <- list.peak[jj,]
  
  objname<-paste("A_",ppeak$chr,"_",ppeak$Ppos,sep="")
  data(list=objname)
  geno <- get(objname)
  geno <- geno[,c("ril", "AA1","AA2","AA3","AA4","AA5","AA6","AA7","AA8")]
  lowset <- geno %>% filter(ril %in% rrs[rrs$Pool==rrcode[(cc+1)],'patRIL'])
  highset <- geno %>% filter(ril %in% rrs[rrs$Pool==rrcode[cc],'patRIL'])
  ppeak[,43:50]<-round(colSums(lowset[,2:9]))
  ppeak[,51:58]<-round(colSums(highset[,2:9]))
  list.peak[jj,]<-ppeak
}
cc<-cc+1
ci.peak[[kk]]<-list.peak
}

save(ci.peak, file ="../ProcessedData/Peaks_wCIs.rda")

```

# Get list of genes under peaks

```{r}
load(file = "../../LearnMemTolRnaseq/DEseq_LearnMemTol/Data/LearnresSVOrder.Rda")
str(LearnresSVorder)
LearnresSVorder$FBgn <- rownames(LearnresSVorder)
Learn_sig_genes <- as.data.frame(LearnresSVorder)

load(file = "../../LearnMemTolRnaseq/DEseq_LearnMemTol/Data/MemresSVOrder.Rda")
str(MemresSVorder)
MemresSVorder$FBgn <- rownames(MemresSVorder)
colnames(MemresSVorder)

Mem_sig_genes <- as.data.frame(MemresSVorder)

gene_map_table <- read_lines(file = "../ProcessedData/gene_map_table_fb_2018_04.tsv",
                skip = 6) %>% 
  str_split(pattern = "\t", simplify = TRUE) %>% 
  as_tibble() %>% 
  filter(V1 == "Dmel")

colnames(gene_map_table) <- c('spp','gname','FBgn','v3','cyt','pos')



gene_map_table$chr <- str_split(gene_map_table$pos, ":",simplify=TRUE)[,1]
temp.s1 <- str_split(gene_map_table$pos, ":",simplify=TRUE)[,2] %>% 
  str_split(fixed(".."),simplify=TRUE)
gene_map_table$startp <- as.numeric(temp.s1[,1])

gene_map_table$stopp <- as.numeric(str_split(temp.s1[,2],fixed("("),simplify = TRUE)[,1])

Learn_gene_list_merged <- merge(Learn_sig_genes, gene_map_table, by="FBgn")
Mem_gene_list_merged <- merge(Mem_sig_genes, gene_map_table, by="FBgn")

Learn_gene_list_sub <- Learn_gene_list_merged[c(1,3,6,7,9,13:15)]
colnames(Learn_gene_list_sub)

Mem_gene_list_sub <- Mem_gene_list_merged[c(1,3,6,7,9,13:15)]
colnames(Mem_gene_list_sub)

#Learning
foc.peak <- ci.peak[[1]][1,]

learn.list <-vector(mode='list', length=nrow(ci.peak$LearnPowerTrans))

Learn_totdf<-data.frame(NULL)
for(row in 1:nrow(ci.peak$LearnPowerTrans)) 
  
  {
  
  foc.peak<-ci.peak[['LearnPowerTrans']][row,]
  lw <- (which(Learn_gene_list_sub$chr==foc.peak$chr & 
                 ((Learn_gene_list_sub$startp <= foc.peak$upR6 & Learn_gene_list_sub$stopp >= foc.peak$lpR6))))  
  
  Learn_genes_under_peak<-Learn_gene_list_sub[lw,]
  learn.list[[row]] <-Learn_genes_under_peak
  
 }


#Memory 

Mem_foc_peak <- ci.peak[["Memory_Mean"]][1,]

Mem.list <-vector(mode='list', length=nrow(ci.peak$Memory_Mean))

Mem_totdf<-data.frame(NULL)
for(row in 1:nrow(ci.peak$Memory_Mean)) 
  
  {
  
  foc.peak<-ci.peak[['Memory_Mean']][row,]
  mw <- (which(Mem_gene_list_sub$chr==foc.peak$chr & 
                 ((Mem_gene_list_sub$startp <= foc.peak$upR6 & Mem_gene_list_sub$stopp >= foc.peak$lpR6))))  
  
 
  Mem_genes_under_peak <- Mem_gene_list_sub[mw,]
  
  Mem.list[[row]] <-Mem_genes_under_peak
  
}

```



# For each peak, get means and compare to eQTL means. Also compare to genes in intervals from cohort expression expt.

```{r}
data("positionlist_wgenetic")

#files from old proj
load(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/goodpeaksinfo_matrix.rda")
load(file="/home/kingeg/Archived/wfitchNOV2015/Fheads_eQTLs/eQTLs/AbyB/Analysis/NewRMA/QTLmapping/means_wcorrected.rda")

effects <- merge(effects, decode.n, by='ID',sort=FALSE, all.x=TRUE)

for(jj in 1:length(learn.list))
{

ll.foc <- learn.list[[jj]]
pp.foc <- ci.peak[[1]][jj,]

mw <- (which(pinfo$chr==pp.foc$chr & 
                 ((pinfo$peaklpB <= pp.foc$up & pinfo$peakupB >= pp.foc$lp))))  
  
ee <- effects[effects$ID %in% pinfo$ID[mw],]

f.high <- which(apply(rbind(as.numeric(pp.foc[,35:42]), as.matrix(ee[,34:41])),2, min) > 4)

qtl.e <- pp.foc[,19:26]
eqtl.e <- ee[,2:9]

qtl.e <- qtl.e[,f.high]
eqtl.e <- eqtl.e[,f.high]

cors.effs <- data.frame('FBgn'= ee$FBgn, 'corEff'= apply(eqtl.e, 1, function(x) cor(x,as.numeric(qtl.e))), stringsAsFactors = FALSE)

ll.all <- merge(ll.foc, cors.effs, by='FBgn', all=TRUE)

ll.all <- merge(ll.all, cors.lm, by='FBgn', all.x=TRUE)
learn.list[[jj]] <- ll.all
}

save(learn.list,file="../ProcessedData/Learn_genes_under_peak.rda")

for(jj in 1:length(Mem.list))
{

ll.foc <- Mem.list[[jj]]
pp.foc <- ci.peak[[1]][jj,]

mw <- (which(pinfo$chr==pp.foc$chr & 
                 ((pinfo$peaklpB <= pp.foc$up & pinfo$peakupB >= pp.foc$lp))))  
  
ee <- effects[effects$ID %in% pinfo$ID[mw],]

f.high <- which(apply(rbind(as.numeric(pp.foc[,35:42]), as.matrix(ee[,34:41])),2, min) > 4)

qtl.e <- pp.foc[,19:26]
eqtl.e <- ee[,2:9]

qtl.e <- qtl.e[,f.high]
eqtl.e <- eqtl.e[,f.high]

cors.effs <- data.frame('FBgn'= ee$FBgn, 'corEff'= apply(eqtl.e, 1, function(x) cor(x,as.numeric(qtl.e))), stringsAsFactors = FALSE)

ll.all <- merge(ll.foc, cors.effs, by='FBgn', all=TRUE)

ll.all <- merge(ll.all, cors.lm, by='FBgn', all.x=TRUE)
Mem.list[[jj]] <- ll.all
}

save(Mem.list,file="../ProcessedData/Mem_genes_under_peak.rda")

```

