---
title: "LearnMem_RNAproc_DESeq"
author: "Patricka Williams-Simon"
date: "8/13/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(dplyr)
library(stringr)
library(sva)
library(ggplot2)
library(RColorBrewer)

```

# Generate a list of gtf commands - ref p.92-93 Dunn book

##grep -li assemb *.gtf > ~/MyGithub/BasePop_RNAseq/processed/gtf_list.txt
# searches and list all .gtf which includes assemb*

#Open gtf_list.txt in TextWrangler [command+F]
#In the search window type ^
#In the replace window, type full path i.e: ~/MyGithub/BasePop_RNAseq/scripts/DESeq_scripts/
#Do Replace All
#This gives full path to each of the .gtf

#Alternatively:
#While in processed/ (i.e. where directory "ballgown" is located), run:
#python ../scripts/DESeq_scripts/prepDE.py
#Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`



# Load gene(/transcript) count matrix and labels

```{r}
LMgenes <- read.csv("../Data/LearnMemBallgown/ballgown_eb/LMgene_count_matrix_eb.csv", row.names="gene_id")

phenData <- read.csv("../Data/sample_description.csv", stringsAsFactors = FALSE)



#Separate traits into different obejects and ad id column 

LMpheno <- phenData %>% filter(treatment %in% c("LL", "LH", "ML", "MH" ))

names(LMpheno)[1]<-"ids"
LMpheno <- LMpheno %>%
  select(ids,treatment,replicate)


rownames(LMpheno) <- LMpheno[,1]
LMpheno[,1] <- NULL

##Add batch column
LMpheno$batch <- LMpheno$treatment

LMpheno$batch[LMpheno$treatment=='LL'] <- "A"
LMpheno$batch[LMpheno$treatment=='LH'] <- "A"
LMpheno$batch[LMpheno$treatment=='ML'] <- "B"
LMpheno$batch[LMpheno$treatment=='MH'] <- "B"

# Check all sample IDs in colData are also in genecount and match their orders
all(rownames(LMpheno) %in% colnames(LMgenes))
LMgenes <- LMgenes[, rownames(LMpheno)]
all(rownames(LMpheno) == colnames(LMgenes))

#colData <- pData(phenDat)[,c("treatment","tissue")]
LMpheno <- as.matrix(LMpheno)

save(LMpheno, file="../Data/LMRNAproc_data_eb.rda")

```

# Model with batch, treatment and tissue

```{r}
# Create a DESeqDataSet from count matrix and labels
LMdds <- DESeqDataSetFromMatrix(countData = LMgenes,
colData = LMpheno, design = ~  replicate + treatment)


# Run the default analysis for DESeq2 and generate results table

LMddsDE <- DESeq(LMdds)
LMres <- results(LMddsDE)

#Sort by adjusted p-value and display
(LMresOrdered <- LMres[order(LMres$padj), ])


```

# Control for batch effects

```{r}
LMdds_sva <- estimateSizeFactors(LMdds)
LMdat_sva <- counts(LMdds_sva, normalized=TRUE)
LMcc_c <- rowSums(LMdat_sva)


#remove all genes with 0s for all samples
LMdat_sva <- LMdat_sva[which(LMcc_c>0),]


LMpheno <- as.data.frame(LMpheno)


#fit model
LMmod <- model.matrix(~as.factor(treatment) +
                               as.factor(replicate),
                              data=LMpheno)


#intercept
LMmod0 <- model.matrix(~as.factor(treatment), data=LMpheno) 
#LMmod0 <- model.matrix(~1, data=LMpheno) 

# calculate number of surrogate variables
LMn_sv <- num.sv(LMdat_sva, LMmod, method = "be")

print(c("Calculated number of significant SVs = ", LMn_sv))


LMsvobj <- svaseq(LMdat_sva, LMmod, LMmod0, n.sv=LMn_sv)


#add the SV values to the pheno dataset
LMpheno$SV1 <- LMsvobj$sv[,1]


LMpheno <- as.data.frame(LMpheno)


#ggplot(LMdat.sva, aes(x=SV1, y=SV2)) +
 # geom_point(alpha=1/5, size=3)

```

#DESeq with surrogate variables, treatment and replicate


```{r}
# countData colnames() must be identical to colData rownames()
all(rownames(LMpheno) %in% colnames(LMgenes))
countdata <- LMgenes[, rownames(LMpheno)]
all(rownames(LMpheno) == colnames(LMgenes))


# DESeqDataSet for all variables,

LMsv_dds_ttSS <- DESeqDataSetFromMatrix(countData = LMgenes, 
                                   colData = LMpheno, 
                                   design = ~ SV1 + treatment + replicate)

LMsv_dds_deseq <- DESeq(LMsv_dds_ttSS, fitType = "local") #3 rows did not converge in beta
LMsv_res_dds <- results(LMsv_dds_deseq)
(LMsv_res_dds <- LMsv_res_dds[order(LMsv_res_dds$padj), ])



## Merge with normalized count data
LMsv_res_dds_table <- merge(as.data.frame(LMsv_res_dds), 
                       as.data.frame(counts(LMsv_dds_deseq, normalized=TRUE)), 
                       by="row.names", sort=FALSE)
names(LMsv_res_dds_table)[1] <- "Gene"


save(LMsv_res_dds_table, file = "../Data/LMDEseqSVA_resOrdered_padj.Rda")

write.csv(LMsv_res_dds_table, 
          "../Data/LMDEseqSVA_resOrdered_padj.csv", row.names=TRUE)

load("../Data/LMDEseqSVA_resOrdered_padj.Rda")

# Examine plot of p-values
hist(LMsv_res_dds$pvalue, breaks=50, col="grey")

# Examine independent filtering
metadata(LMsv_res_dds)$filterThreshold

plot(metadata(LMsv_res_dds)$filterNumRej, type="b", xlab="quantiles of baseMean", ylab="number of rejections")
```

