---
title: "LearnMem_RNAproc_DESeq"
author: "Patricka Williams-Simon"
date: "8/13/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(dplyr)
library(stringr)
library(sva)
library(ggplot2)
library(RColorBrewer)

set.seed(38142903)
```

# Generate a list of gtf commands - ref p.92-93 Dunn book

##grep -li assemb *.gtf > ~/MyGithub/BasePop_RNAseq/processed/gtf_list.txt
# searches and list all .gtf which includes assemb*

#Open gtf_list.txt in TextWrangler [command+F]
#In the search window type ^
#In the replace window, type full path i.e: ~/MyGithub/BasePop_RNAseq/scripts/DESeq_scripts/
#Do Replace All
#This gives full path to each of the .gtf

#Alternatively:
#While in processed/ (i.e. where directory "ballgown" is located), run:
#python ../scripts/DESeq_scripts/prepDE.py
#Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`



# Load gene(/transcript) count matrix and labels

```{r}
LearnMemgenecount <- read.csv("../Data/LearnMem_gene_count_matrix.csv", row.names="gene_id")

phenData <- read.csv("../Data/sample_description.csv", stringsAsFactors = FALSE)



#Separate traits into different obejects and ad id column 

LearnMempheno_data <- phenData %>% filter(treatment %in% c("LL", "LH", "ML", "MH" ))

names(LearnMempheno_data)[1]<-"ids"
LearnMempheno_data <- LearnMempheno_data %>%
  select(ids,treatment,replicate)

rownames(LearnMempheno_data) <- LearnMempheno_data[,1]
LearnMempheno_data[,1] <- NULL


# Check all sample IDs in colData are also in genecount and match their orders
all(rownames(LearnMempheno_data) %in% colnames(LearnMemgenecount))
LearnMemgenecount <- LearnMemgenecount[, rownames(LearnMempheno_data)]
all(rownames(LearnMempheno_data) == colnames(LearnMemgenecount))

#colData <- pData(phenDat)[,c("treatment","tissue")]
LearnMempheno_data <- as.matrix(LearnMempheno_data)

```

# Model with treatment and tissue

```{r}
# Create a DESeqDataSet from count matrix and labels
ddsLearnMem <- DESeqDataSetFromMatrix(countData = LearnMemgenecount,
colData = LearnMempheno_data, design = ~ replicate + treatment)


# Run the default analysis for DESeq2 and generate results table

LearnMemddsDE <- DESeq(ddsLearnMem)
LearnMemres <- results(LearnMemddsDE)

#Sort by adjusted p-value and display
(LearnMemresOrdered <- LearnMemres[order(LearnMemres$padj), ])


save(LearnMemddsDE, file = "../Data/LearnMemddsDE.Rda")
save(LearnMemresOrdered, file = "../Data/LearnMemres.Rda")
```

# Control for batch effects

```{r}
LearnMemdds.sva <- estimateSizeFactors(ddsLearnMem)
LearnMemdat.sva <- counts(LearnMemdds.sva, normalized=TRUE)
LearnMemcc.c <- rowSums(LearnMemdat.sva)


#remove all genes with 0s for all samples
LearnMemdat.sva <- LearnMemdat.sva[which(LearnMemcc.c>0),]


#load data with batch numbers
phenData.sva <- read.csv("../Data/sample_description_batch.csv", stringsAsFactors = FALSE)



LearnMempheno_data.sva <- phenData.sva %>% filter(treatment %in% c("LL", "LH", "ML", "MH" ))

#names(LearnMempheno_data.sva)[5]<-"batch"
#cbind(LearnMempheno_data.sva, colnames(dat.sva))


#fit model
LearnMemmod <- model.matrix(~as.factor(treatment) +
                               as.factor(replicate),
                              data=LearnMempheno_data.sva)


#intercept
LearnMemmod0 <- model.matrix(~1+ as.factor(replicate), 
                             data=LearnMempheno_data.sva) 

# calculate number of surrogate variables
LearnMemn.sv <- num.sv(LearnMemdat.sva, LearnMemmod, method = "be")

print(c("Calculated number of significant SVs = ", LearnMemn.sv))


LearnMemsvobj <- svaseq(LearnMemdat.sva, LearnMemmod, LearnMemmod0, LearnMemn.sv=LEarnMemn.sv)


LearnMempheno_data.sva.sva$SV1 <- svobj$sv[,1]
LearnMempheno_data.sva$SV2 <- svobj$sv[,2]

ggplot(LearnMempheno_data.sva, aes(x=SV1, y=SV2)) +
  geom_point(alpha=1/5, size=3)



```


