---
title: "Prep DESeq-worthy counts from StringTie output"
author: "Patricka Williams-Simon"
date: "7/19/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(dplyr)
library(stringr)
library(sva)
library(ggplot2)
```

# Generate a list of gtf commands - ref p.92-93 Dunn book

##grep -li assemb *.gtf > ~/MyGithub/BasePop_RNAseq/processed/gtf_list.txt
# searches and list all .gtf which includes assemb*

#Open gtf_list.txt in TextWrangler [command+F]
#In the search window type ^
#In the replace window, type full path i.e: ~/MyGithub/BasePop_RNAseq/scripts/DESeq_scripts/
#Do Replace All
#This gives full path to each of the .gtf

#Alternatively:
#While in processed/ (i.e. where directory "ballgown" is located), run:
#python ../scripts/DESeq_scripts/prepDE.py
#Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`

# Load gene(/transcript) count matrix and labels

```{r}

ThermTolgenecount <- read.csv("../Data/ThermTol_gene_count_matrix.csv", row.names="gene_id")


phenData <- read.csv("../Data/sample_description.csv", stringsAsFactors = FALSE)


#Separate traits into different obejects and add id column 

Tolpheno_data <- phenData %>% filter(treatment %in% c("TH", "TH_Incu", "TH_Incu_control", "TL","TL_Incu", "TL_Incu_control"  ))

names(Tolpheno_data)[1]<-"ids"
Tolpheno_data <- Tolpheno_data %>%
  select(ids,treatment,replicate)

rownames(Tolpheno_data) <- Tolpheno_data[,1]
Tolpheno_data[,1] <- NULL

#Add pool, condition an batch column
Tolpheno_data$pool <- Tolpheno_data$treatment
Tolpheno_data$condition <- Tolpheno_data$treatment
Tolpheno_data$batch <- Tolpheno_data$treatment


#assign variables o new columns
Tolpheno_data$pool[Tolpheno_data$treatment=='TL'] <- "L"
Tolpheno_data$pool[Tolpheno_data$treatment=='TL_Incu'] <- "L"
Tolpheno_data$pool[Tolpheno_data$treatment=='TL_Incu_control'] <- "L"
Tolpheno_data$pool[Tolpheno_data$treatment=='TH'] <- "H"
Tolpheno_data$pool[Tolpheno_data$treatment=='TH_Incu'] <- "H"
Tolpheno_data$pool[Tolpheno_data$treatment=='TH_Incu_control'] <- "H"

Tolpheno_data$condition[Tolpheno_data$treatment=='TL'] <- "RT"
Tolpheno_data$condition[Tolpheno_data$treatment=='TL_Incu'] <- "Incu"
Tolpheno_data$condition[Tolpheno_data$treatment=='TL_Incu_control'] <- "RT"
Tolpheno_data$condition[Tolpheno_data$treatment=='TH'] <- "RT"
Tolpheno_data$condition[Tolpheno_data$treatment=='TH_Incu'] <- "Incu"
Tolpheno_data$condition[Tolpheno_data$treatment=='TH_Incu_control'] <- "RT"

Tolpheno_data$batch[Tolpheno_data$treatment=='TL'] <- "1"
Tolpheno_data$batch[Tolpheno_data$treatment=='TL_Incu'] <- "2"
Tolpheno_data$batch[Tolpheno_data$treatment=='TL_Incu_control'] <- "2"
Tolpheno_data$batch[Tolpheno_data$treatment=='TH'] <- "1"
Tolpheno_data$batch[Tolpheno_data$treatment=='TH_Incu'] <- "2"
Tolpheno_data$batch[Tolpheno_data$treatment=='TH_Incu_control'] <- "2"



#order thermtol dataset 

Tolpheno_data <- Tolpheno_data[order (row.names(Tolpheno_data)),]

# Check all sample IDs in colData are also in genecount and match their orders

all(rownames(Tolpheno_data) %in% colnames(ThermTolgenecount))
genecount2 <- ThermTolgenecount[, rownames(Tolpheno_data)]
all(rownames(Tolpheno_data) == colnames(ThermTolgenecount))


#colData <- pData(phenDat)[,c("treatment","tissue")]
Tolpheno_data <- as.matrix(Tolpheno_data)


save(Tolpheno_data, file="../Data/ThermRNAproc_data.rda")


```

# Model with treatment 

```{r}

load(file="../Data/ThermRNAproc_data.rda")

# Create a DESeqDataSet from count matrix and labels
ddsThermTol <- DESeqDataSetFromMatrix(countData = ThermTolgenecount,
colData = Tolpheno_data, design = ~ batch + pool + condition)


# Run the default analysis for DESeq2 and generate results table

ThermTolddsDE <- DESeq(ddsThermTol)
ThermTolres <- results(ThermTolddsDE)

#Sort by adjusted p-value and display

(ThermTolresOrdered <- ThermTolres[order(ThermTolres$padj), ])


save(ThermTolddsDE, file = "../Data/ThermTolddsDE.Rda")
save(ThermTolresOrdered, file = "../Data/ThermTolres.Rda")

```


```{r}
# DESeqDataSet for all variables, no interaction
ThermToldds.ttSS <- DESeqDataSetFromMatrix(countData = ThermTolgenecount, 
                                   colData = Tolpheno_data, 
                                   design = ~ pool + condition)


dds_deseq <- DESeq(ThermToldds.ttSS)
res.dds <- results(dds_deseq)
(res.dds <- res.dds[order(res.dds$padj), ])
```


# Control for batch effects

```{r}
ThermToldds.sva <- estimateSizeFactors(ddsThermTol)
ThermToldat.sva <- counts(ThermToldds.sva, normalized=TRUE)
ThermTolcc.c <- rowSums(ThermToldat.sva)

#remove all genes with 0s for all samples
ThermToldat.sva <- ThermToldat.sva[which(ThermTolcc.c>0),]

#set data as dataframe
Tolpheno_data <- as.data.frame(Tolpheno_data)
ThermToldat.sva <- as.data.frame(ThermToldat.sva)


#fit model
ThermTolmod <- model.matrix(~ as.factor(batch) + 
                              as.factor(pool) + 
                              as.factor(condition), 
                              data=Tolpheno_data)
#intercept
ThermTolmod0 <- model.matrix(~1, data=ThermToldat.sva) 


# calculate number of surrogate variables
ThermToln.sv <- num.sv(ThermToldat.sva, ThermTolmod)

#ThermToln.sv <- num.sv(ThermToldat.sva, ThermTolmod, method = "be", B =5)
#ThermToln.sv <- num.sv(ThermToldat.sva, ThermTolmod0)

print(c("Calculated number of significant SVs = ", ThermToln.sv))

ThermTolsvobj <- svaseq(ThermToldat.sva, ThermTolmod, ThermTolmod0, n.sv=ThermToln.sv) 

#ThermToldat.sva$SV1 <- ThermTolsvobj$sv[,1]
#ThermToldat.sva$SV2 <- ThermTolsvobj$sv[,2]

ggplot(ThermToldat.sva, aes(x=SV1, y=SV2)) +
geom_point(alpha=1/5, size=3)

```

