---
title: "Prep DESeq-worthy counts from StringTie output"
author: "Patricka Williams-Simon"
date: "7/19/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(dplyr)
library(stringr)
library(sva)
library(ggplot2)
```

# Generate a list of gtf commands - ref p.92-93 Dunn book

##grep -li assemb *.gtf > ~/MyGithub/BasePop_RNAseq/processed/gtf_list.txt
# searches and list all .gtf which includes assemb*

#Open gtf_list.txt in TextWrangler [command+F]
#In the search window type ^
#In the replace window, type full path i.e: ~/MyGithub/BasePop_RNAseq/scripts/DESeq_scripts/
#Do Replace All
#This gives full path to each of the .gtf

#Alternatively:
#While in processed/ (i.e. where directory "ballgown" is located), run:
#python ../scripts/DESeq_scripts/prepDE.py
#Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`

# Load gene(/transcript) count matrix and labels

```{r}

# Read in data frame and fill in empty spaces in column 2 with value in 1
TTgenes <- read.csv("../Data/ThermTolBallgown/TTgene_count_matrix1.csv", stringsAsFactors = FALSE)

TTgenes$gene_id <- ifelse(TTgenes$gene_id == "", TTgenes$gene_id0, TTgenes$gene_id) 
TTgenes <- TTgenes[c(2, 1, 3:33)]

#find duplicated rows

#w <- which(duplicated(TTgenes[,1]))
#rownames(TTgenes, c(13988, 14695))

###
###
####
######change row name, b/c it was deplicated with row 14695

#TTgenes[13988,]<-"FBgn0002781.1"

#it's gone!
#rownames(TTgenes) <- c("13988")

#duplicated rows
#TTdup = rownames(TTgenes)
#TTdup[duplicated(TTdup$gene_id)]


#delete first 3 row name
rownames(TTgenes) <- TTgenes[,1]
TTgenes[,2:3] <- NULL

# Reformat column names
names(TTgenes) = sub("\\.","",names(TTgenes)) #regex escape the escape
TTgenes <- as.matrix(TTgenes)


#load phenotype data


PhenoData <- read.csv("../Data/sample_description.csv", stringsAsFactors = FALSE)


#Separate traits into different obejects and add id column 

TTpheno <- PhenoData %>% filter(treatment %in% c("TH", "TH_Incu", "TH_Incu_control", "TL","TL_Incu", "TL_Incu_control"  ))

names(TTpheno)[1]<-"ids"
TTpheno <- TTpheno %>%
  select(ids,treatment,replicate)

rownames(TTpheno) <- TTpheno[,1]
TTpheno[,1] <- NULL

#Add pool, condition an batch column
TTpheno$pool <- TTpheno$treatment
TTpheno$condition <- TTpheno$treatment
TTpheno$batch <- TTpheno$treatment


#assign variables o new columns
TTpheno$pool[TTpheno$treatment=='TL'] <- "L"
TTpheno$pool[TTpheno$treatment=='TL_Incu'] <- "L"
TTpheno$pool[TTpheno$treatment=='TL_Incu_control'] <- "L"
TTpheno$pool[TTpheno$treatment=='TH'] <- "H"
TTpheno$pool[TTpheno$treatment=='TH_Incu'] <- "H"
TTpheno$pool[TTpheno$treatment=='TH_Incu_control'] <- "H"

TTpheno$condition[TTpheno$treatment=='TL'] <- "RT"
TTpheno$condition[TTpheno$treatment=='TL_Incu'] <- "Incu"
TTpheno$condition[TTpheno$treatment=='TL_Incu_control'] <- "RT"
TTpheno$condition[TTpheno$treatment=='TH'] <- "RT"
TTpheno$condition[TTpheno$treatment=='TH_Incu'] <- "Incu"
TTpheno$condition[TTpheno$treatment=='TH_Incu_control'] <- "RT"

TTpheno$batch[TTpheno$treatment=='TL'] <- "1"
TTpheno$batch[TTpheno$treatment=='TL_Incu'] <- "2"
TTpheno$batch[TTpheno$treatment=='TL_Incu_control'] <- "2"
TTpheno$batch[TTpheno$treatment=='TH'] <- "1"
TTpheno$batch[TTpheno$treatment=='TH_Incu'] <- "2"
TTpheno$batch[TTpheno$treatment=='TH_Incu_control'] <- "2"



#order thermtol dataset 

TTpheno <- TTpheno[order (row.names(TTpheno)),]

# Check all sample IDs in colData are also in genecount and match their orders

all(rownames(TTpheno) %in% colnames(TTgenes))
genecount2 <- TTgenes[, rownames(TTpheno)]
all(rownames(TTpheno) == colnames(TTgenes))


#colData <- pData(phenDat)[,c("treatment","tissue")]
TTpheno <- as.matrix(TTpheno)


save(TTpheno, file="../Data/ThermRNAproc_data.rda")


```

# Model with treatment 

```{r}

load(file="../Data/ThermRNAproc_data.rda")

# Create a DESeqDataSet from count matrix and labels
ddsThermTol <- DESeqDataSetFromMatrix(countData = ThermTolgenecount,
colData = TTpheno, design = ~ batch + pool + condition)


# Run the default analysis for DESeq2 and generate results table

ThermTolddsDE <- DESeq(ddsThermTol)
ThermTolres <- results(ThermTolddsDE)

#Sort by adjusted p-value and display

(ThermTolresOrdered <- ThermTolres[order(ThermTolres$padj), ])


save(ThermTolddsDE, file = "../Data/ThermTolddsDE.Rda")
save(ThermTolresOrdered, file = "../Data/ThermTolres.Rda")

```


```{r}
# DESeqDataSet for all variables, no interaction
ThermToldds.ttSS <- DESeqDataSetFromMatrix(countData = ThermTolgenecount, 
                                   colData = TTpheno, 
                                   design = ~ pool + condition)


dds_deseq <- DESeq(ThermToldds.ttSS)
res.dds <- results(dds_deseq)
(res.dds <- res.dds[order(res.dds$padj), ])
```


# Control for batch effects

```{r}
ThermToldds.sva <- estimateSizeFactors(ddsThermTol)
ThermToldat.sva <- counts(ThermToldds.sva, normalized=TRUE)
ThermTolcc.c <- rowSums(ThermToldat.sva)

#remove all genes with 0s for all samples
ThermToldat.sva <- ThermToldat.sva[which(ThermTolcc.c>0),]

#set data as dataframe
TTpheno <- as.data.frame(TTpheno)
ThermToldat.sva <- as.data.frame(ThermToldat.sva)


#fit model
ThermTolmod <- model.matrix(~ as.factor(batch) + 
                              as.factor(pool) + 
                              as.factor(condition), 
                              data=TTpheno)
#intercept
ThermTolmod0 <- model.matrix(~1, data=ThermToldat.sva) 


# calculate number of surrogate variables
ThermToln.sv <- num.sv(ThermToldat.sva, ThermTolmod)

#ThermToln.sv <- num.sv(ThermToldat.sva, ThermTolmod, method = "be", B =5)
#ThermToln.sv <- num.sv(ThermToldat.sva, ThermTolmod0)

print(c("Calculated number of significant SVs = ", ThermToln.sv))

ThermTolsvobj <- svaseq(ThermToldat.sva, ThermTolmod, ThermTolmod0, n.sv=ThermToln.sv) 

#ThermToldat.sva$SV1 <- ThermTolsvobj$sv[,1]
#ThermToldat.sva$SV2 <- ThermTolsvobj$sv[,2]

ggplot(ThermToldat.sva, aes(x=SV1, y=SV2)) +
geom_point(alpha=1/5, size=3)

```

