---
title: "Prep DESeq-worthy counts from StringTie output"
author: "Patricka Williams-Simon"
date: "7/19/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(DESeq2)
library(dplyr)
library(stringr)
library(sva)
library(ggplot2)
```

# Generate a list of gtf commands - ref p.92-93 Dunn book

##grep -li assemb *.gtf > ~/MyGithub/BasePop_RNAseq/processed/gtf_list.txt
# searches and list all .gtf which includes assemb*

#Open gtf_list.txt in TextWrangler [command+F]
#In the search window type ^
#In the replace window, type full path i.e: ~/MyGithub/BasePop_RNAseq/scripts/DESeq_scripts/
#Do Replace All
#This gives full path to each of the .gtf

#Alternatively:
#While in processed/ (i.e. where directory "ballgown" is located), run:
#python ../scripts/DESeq_scripts/prepDE.py
#Two output matrices are written: `transcript_count_matrix.csv` and `gene_count_matrix.csv`

# Load gene(/transcript) count matrix and labels

```{r}

# Read in data frame and fill in empty spaces in column 2 with value in 1
TTgenes <- read.csv("../Data/ThermTolBallgown/ballgown_eb/TTgene_count_matrix_eb.csv", stringsAsFactors = FALSE)


#delete first 3 row name, so that data set macthes up with pheno data.
rownames(TTgenes) <- TTgenes[,1]
TTgenes[,1] <- NULL


TTgenes <- as.matrix(TTgenes)


#load phenotype data

PhenoData <- read.csv("../Data/sample_description.csv", stringsAsFactors = FALSE)


#Separate traits into different obejects and add id column 

TTpheno <- PhenoData %>% filter(treatment %in% c("TH", "TH_Incu", "TH_Incu_control", "TL","TL_Incu", "TL_Incu_control"  ))

names(TTpheno)[1]<-"ids"
TTpheno <- TTpheno %>%
  select(ids,treatment,replicate)

rownames(TTpheno) <- TTpheno[,1]
TTpheno[,1] <- NULL

#Add pool, condition an batch column
TTpheno$pool <- TTpheno$treatment
TTpheno$condition <- TTpheno$treatment
TTpheno$batch <- TTpheno$treatment


#assign variables o new columns
TTpheno$pool[TTpheno$treatment=='TL'] <- "L"
TTpheno$pool[TTpheno$treatment=='TL_Incu'] <- "L"
TTpheno$pool[TTpheno$treatment=='TL_Incu_control'] <- "L"
TTpheno$pool[TTpheno$treatment=='TH'] <- "H"
TTpheno$pool[TTpheno$treatment=='TH_Incu'] <- "H"
TTpheno$pool[TTpheno$treatment=='TH_Incu_control'] <- "H"

TTpheno$condition[TTpheno$treatment=='TL'] <- "RT"
TTpheno$condition[TTpheno$treatment=='TL_Incu'] <- "Incu"
TTpheno$condition[TTpheno$treatment=='TL_Incu_control'] <- "RT"
TTpheno$condition[TTpheno$treatment=='TH'] <- "RT"
TTpheno$condition[TTpheno$treatment=='TH_Incu'] <- "Incu"
TTpheno$condition[TTpheno$treatment=='TH_Incu_control'] <- "RT"

TTpheno$batch[TTpheno$treatment=='TL'] <- "1"
TTpheno$batch[TTpheno$treatment=='TL_Incu'] <- "2"
TTpheno$batch[TTpheno$treatment=='TL_Incu_control'] <- "2"
TTpheno$batch[TTpheno$treatment=='TH'] <- "1"
TTpheno$batch[TTpheno$treatment=='TH_Incu'] <- "2"
TTpheno$batch[TTpheno$treatment=='TH_Incu_control'] <- "2"



#order thermtol dataset 

TTpheno <- TTpheno[order (row.names(TTpheno)),]

# Check all sample IDs in colData are also in genecount and match their orders

all(rownames(TTpheno) %in% colnames(TTgenes))
genecount2 <- TTgenes[, rownames(TTpheno)]
all(rownames(TTpheno) == colnames(TTgenes))


#colData <- pData(phenDat)[,c("treatment","tissue")]
TTpheno <- as.matrix(TTpheno)


save(TTpheno, file="../Data/ThermRNAproc_data.rda")


```

# Model with pool and condition 

```{r}

load(file="../Data/ThermRNAproc_data.rda")

# Create a DESeqDataSet from count matrix and labels
TTdds <- DESeqDataSetFromMatrix(countData = TTgenes,
colData = TTpheno, design = ~  pool + condition)


# Run the default analysis for DESeq2 and generate results table

TTDE <- DESeq(TTdds)
TTres <- results(TTDE)

#Sort by adjusted p-value and display

(TTresOrdered <- TTres[order(TTres$padj), ])

```

#Model with batch, pool and condition
```{r}
# DESeqDataSet for all variables, no interaction
TTdds2 <- DESeqDataSetFromMatrix(countData = TTgenes, 
                                   colData = TTpheno, 
                                   design = ~ batch + pool + condition)


TTdds_deseq2 <- DESeq(TTdds2)
TTres_dds2 <- results(TTdds_deseq2)
(TTresOrder2 <- TTres_dds2[order(TTres_dds2$padj), ])


```


# Control for batch effects

```{r}


TTdds_sva <- estimateSizeFactors(TTdds)
TTdat_sva <- counts(TTdds_sva, normalized=TRUE)
TTcc_c <- rowSums(TTdat_sva)

#remove all genes with 0s for all samples
TTdat_sva <- TTdat_sva[which(TTcc_c>0),]


TTpheno <- as.data.frame(TTpheno)

#fit model
TTmod <- model.matrix(~ as.factor(batch) + 
                        as.factor(pool) +
                        as.factor(condition), 
                         data=TTpheno)
#intercept
TTmod0 <- model.matrix(~ as.factor(batch), data=TTpheno) 


# calculate number of surrogate variables

TTn_sv <- num.sv(TTdat_sva, TTmod)

print(c("Calculated number of significant SVs = ", TTn_sv))

TTsv_obj <- svaseq(TTdat_sva, TTmod, TTmod0, n.sv=TTn_sv) 

#add SVs to phenotype dataset
TTpheno$SV1 <- TTsv_obj$sv[,1]
TTpheno$SV2 <- TTsv_obj$sv[,2]


head(TTdat_sva, 3)

TTdat_sva <- as.data.frame(TTdat_sva)

ggplot(TTpheno, aes(x=SV1, y=SV2)) +
geom_point(alpha=1/5, size=3)

```


#DESeq with surrogate variables


```{r}
# countData colnames() must be identical to colData rownames()
all(rownames(TTpheno) %in% colnames(TTgenes))
countdata <- TTgenes[, rownames(TTpheno)]
all(rownames(TTpheno) == colnames(TTgenes))


# DESeqDataSet for all variables,

TTsv_dds_ttSS <- DESeqDataSetFromMatrix(countData = TTgenes, 
                                   colData = TTpheno, 
                                   design = ~ SV1 + SV2 + batch + pool + condition)

TTsv_dds_deseq <- DESeq(TTsv_dds_ttSS, fitType = "local")
TTsv_res_dds <- results(TTsv_dds_deseq)
(TTsv_res_dds <- TTsv_res_dds[order(TTsv_res_dds$padj), ])



## Merge with normalized count data
TTsv_res_dds_table <- merge(as.data.frame(TTsv_res_dds), 
                       as.data.frame(counts(TTsv_dds_deseq, normalized=TRUE)), 
                       by="row.names", sort=FALSE)
names(TTsv_res_dds_table)[1] <- "Gene"


write.csv(TTsv_res.dds_table, 
          "../Data/TTDEseqSVA_resOrdered_padj.csv", row.names=TRUE)

load("../Data/TTDEseqSVA_resOrdered_padj.Rda")

# Examine plot of p-values
hist(TTsv_res_dds$pvalue, breaks=50, col="grey")

# Examine independent filtering
metadata(TTsv_res_dds)$filterThreshold

plot(metadata(TTsv_res_dds)$filterNumRej, type="b", xlab="quantiles of baseMean", ylab="number of rejections")
```

